packages/recommend-vdom/src/types/RecommendComponentProps.ts

```diff
@@ -15,7 +15,7 @@ export type HeaderComponentProps<TObject> = Renderer & ComponentProps<TObject>;
 
 export type ComponentProps<TObject> = {
   classNames: RecommendClassNames;
-  recommendations: TObject[];
+  recommendations: TObject[] | Array<FacetEntry<TObject>>;
```

Comment:
We should prob pass `Array<FacetEntry<TObject>>` at call site for `TObject`. Otherwise we'll have wrong types for other models which would assume that they can receive `Array<FacetEntry<TObject>>` recommendations.

Reply:
Good catch !